capa/features/extractors/binexport2/insn.py

```diff
@@ -0,0 +1,443 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import io
+import logging
+from typing import List, Tuple, Iterator
+
+import pefile
+from elftools.elf.elffile import ELFFile
+
+import capa.features.extractors.helpers
+import capa.features.extractors.strings
+from capa.features.insn import API, Number, Mnemonic, OperandNumber
+from capa.features.common import Bytes, String, Feature, Characteristic
+from capa.features.address import Address, AbsoluteVirtualAddress
+from capa.features.extractors.binexport2 import AnalysisContext, FunctionContext, BasicBlockContext, InstructionContext
+from capa.features.extractors.base_extractor import BBHandle, InsnHandle, FunctionHandle
+from capa.features.extractors.binexport2.binexport2_pb2 import BinExport2
+
+logger = logging.getLogger(__name__)
+
+
+def extract_insn_api_features(fh: FunctionHandle, _bbh: BBHandle, ih: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
+    fhi: FunctionContext = fh.inner
+    ii: InstructionContext = ih.inner
+
+    be2 = fhi.ctx.be2
+    idx = fhi.ctx.idx
+    analysis = fhi.ctx.analysis
+
+    instruction = be2.instruction[ii.instruction_index]
+
+    if not instruction.call_target:
+        return
+
+    for call_target_address in instruction.call_target:
+        if call_target_address in analysis.thunks:
+            vertex_index = analysis.thunks[call_target_address]
+        elif call_target_address not in idx.vertex_index_by_address:
+            continue
+        else:
+            vertex_index = idx.vertex_index_by_address[call_target_address]
+
+        vertex = be2.call_graph.vertex[vertex_index]
+        if not vertex.HasField("mangled_name"):
+            continue
+
+        yield API(vertex.mangled_name), ih.address
```

Comment:
in tests `vertex.mangled_name` seems to start with `.` e.g. `.fopen`

`vertex.demangled_name` is `_fopen` in this case, maybe we can use that (and then apply the remove leading `_` logic)